### ISSUE TITLE
Operator failing when creating a secret "Invalid value: \"created by\""

### ISSUE THREAD

**Issue by nicolasdonoso** [1Password/onepassword-operator#48]
### Your environment

Operator Version: 0.0.1

Connect Server Version: 1.2.0

Kubernetes Version: v1.19.9

## What happened?
When creating a OnePasswordItem, the custom resource gets created but then the operator cannot create the secret as expected. These are the logs that the operator throws:

```
{"level":"info","ts":1624901800.3914187,"logger":"controller_onepassworditem","msg":"Reconciling OnePasswordItem","Request.Namespace":"dev","Request.Name":"onepwd"}
{"level":"info","ts":1624901802.519541,"msg":"Creating Secret onepwd at namespace 'dev'"}
{"level":"error","ts":1624901802.52166,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"onepassworditem-controller","request":"dev/onepwd","error":"Secret \"onepwd\" is invalid: [data[created by]: Invalid value: \"created by\": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+'), data[key base64]: Invalid value: \"key base64\": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')]","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/workspace/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90"}
```

This is how the OnePasswordItem looks:

```
apiVersion: onepassword.com/v1
kind: OnePasswordItem
metadata:
  name: onepwd
  namespace: dev
spec:
  itemPath: "vaults/<vault name>/items/<item name>" 
```

## What did you expect to happen?
Once the OnePasswordItem is created a secret should be created so then I would be able to see it by running the following command:

`kubectl get secret onepwd -n dev`

## Steps to reproduce
1. Follow [this](https://github.com/1Password/onepassword-operator#setup) and then create the OnePasswordItem mentioned above

**Comment by florisvdg** [comment]
Kubernetes doesn't allow that secrets have keys with spaces. For now the workaround would be to replace the spaces in your 1Password item fields with a dash or underscore character. We'll look into automating this on the operator-side as well.

**Comment by nicolasdonoso** [comment]
The secret value itself is not the issue, it is a label being added to the secret yaml that has an space on it when creating it from a OnePasswordItem.

`[data[created by]: Invalid value: \"created by\":`

This is how the OnePasswordItem yaml looks like:

```
apiVersion: onepassword.com/v1
kind: OnePasswordItem
metadata:
  name: onepwd
  namespace: dev
spec:
  itemPath: "vaults/<vault-name>/items/<item-id>" 
```

**Comment by florisvdg** [comment]
Yes, but if you open your 1Password app and view this item, does it have a field in there called `created by`?

**Comment by verkaufer** [comment]
@nicolasdonoso Dropping in to let you know a fix is in progress. We don't have a projected release date for the next version with the fix, but you can watch #58 for updates. Thanks again for your patience.
